### PR TITLE
Async for issue #11

### PR DESCRIPTION
**Pull request by 4rumprom**
Hi itsjafer,

You mentioned code readability as the main issue and not wanting to support sync & async simultaneously.
My initial thought was to to cancel the pull request but I then realized it was pretty straight forward to go fully async.

This will solve issue #11 and is backwards compatible.

I ran this code on streamlit which is incompatible with sync and also tested it with github actions to make sure async works. (cf. test.yml under workflows).

### CODE REVIEW

File: schwab_api/authentication.py
Comment (tristanpenman) [inline schwab_api/authentication.py:68]: This variable seems to be unused

File: schwab_api/authentication.py
Comment (tristanpenman) [inline schwab_api/authentication.py:99]: This line may be redundant now

Comment (4rumprom) [reply schwab_api/authentication.py:99]: Yes it is.

File: schwab_api/authentication.py
Comment (4rumprom) [reply schwab_api/authentication.py:68]: I'm surprised it worked without it but it's better to use it for robustness. I added it in.

File: schwab_api/authentication.py
Comment (itsjafer) [inline schwab_api/authentication.py:37]: nit: typo

**Review by itsjafer — APPROVED** [review]
The code itself looks good to me. I'm happy to merge this in, see if the tests work as expected, and then push to PyPi.

Does it make sense for this to a major version bump? i.e. from 0.3.X to 0.4.X because of the async support

**Comment by 4rumprom** [comment]
> Does it make sense for this to a major version bump? i.e. from 0.3.X to 0.4.X because of the async support

I'd think so. Especially since the option strategies would be released with this version as well.

**Comment by 4rumprom** [comment]
I am also experiencing login issues today.
It seems that the url to place trades has changed, at least on my end.
I have made a change to the login to ensure more robustness by looking for the keywords "app/trade" in the url instead of an exact url.

File: schwab_api/authentication.py
Comment (4rumprom) [inline schwab_api/authentication.py:95]: I am also experiencing login issues today.
It seems that the url to place trades has changed, at least on my end.
I have made a change to the login to ensure more robustness by looking for the keywords "app/trade" in the url instead of an exact url.

**Comment by 4rumprom** [comment]
I am also experiencing login issues today.
It seems that the url to place trades has changed, at least on my end.
I have made a change to the login to ensure more robustness by looking for the keywords "app/trade" in the url instead of an exact url.

`await self.page.wait_for_url(".*app/trade.*") # Making it more robust than specigying an exact url which may change.`

File: schwab_api/authentication.py
Comment (4rumprom) [reply schwab_api/authentication.py:96]: seems like re.compile(r"app/trade") should be passed as argument instead of the regular expression string itself.

**Comment by 4rumprom** [comment]
Hi Jafer,
Is there something missing on my end to complete this?
Tks,
Romain

**Comment by itsjafer** [comment]
Are you not able to merge this in? I thought once I approve the PR, it can be merged

**Comment by 4rumprom** [comment]
Not 100% sure. This is all pretty new to me. 
It does say that only those with write access to this repository can merge pull requests.

image

**Pull request merged**
